### ISSUE TITLE
Multiple subscriptions causing troubles

### ISSUE THREAD

**Issue by cjblomqvist** [angular/angularfire2#215]
Scenario in Angular:
**Service A:**
```
constructor() {
  this.list = this.angularfire.list('/myList');
}
```

**Component A:**
```
onInit() {
  this.list = this.serviceA.list;
}
```

```
<div *ngFor="let item of list | async">
  <componentB />
</div>
```

**Component B:**
`<componentC />`

**Component C:**
```
onInit() {
  this.list = this.serviceA.list;
}
```

`<div *ngFor="let item of list | async">TEST</div>`

Then you update one of the items in list (e.g. through Firebase admin web interface). This will cause the following:
1) Component C will be destroyed
2) Which causes the async pipe to unsubscribe to the this.list Observable.
3) Due to https://github.com/angular/angularfire2/blob/master/src/utils/firebase_list_factory.ts#L139 - all listeners/subscriptions are cancelled to the ref. Since we have two listeners/subscriptions (one at Component C which should cancel, and one at Component A which should NOT cancel), this will break the app (more specifically, Component A's subscription will cancel which it shouldn't).

I'll send a PR ASAP to fix this.

**Comment by davideast** [comment]
Hey @cjblomqvist!

We're in the middle of upgrading to the Firebase 3.0 SDK, see #200. This is a big undertaking and we won't be able to merge this in just yet.

Can you provide a plnkr of this problem so we can take a look at it ourselves?

Also, in the future please wait until discussion before sending a PR. It's good to have conversation around the issue before writing any code.

**Comment by cjblomqvist** [comment]
Hey @davideast 

I don't have much time at the moment for a plnkr - I'll get back to you though ASAP I find the time.

Regarding the PR. I had to do the fix for a project we're doing (where we encounter the issue), hence the PR part was very small. Also, since this was a pure bug fix which seemed very obvious once you understand the problem, I did not feel the need to check too much before submitting. Anyway, despite this I'm very open for a discussion and I can understand your motive behind it :-)

**Comment by davideast** [comment]
@cjblomqvist  The 3.0 PR is in! I'd like to come back and address this issue now if you have the time. Let me know if you can create that plnkr so I can reproduce the problem myself.

**Comment by ivoviz** [comment]
@davideast Is there a base plnkr that we can use? I'm having a similar problem with `FirebaseObjectObservable` (not so drastic that it destroys the component, but it still makes angularfire unusable in production) and I would gladly reproduce it, but making angularfire work in plnkr is... well, meh.

**Comment by davideast** [comment]
Fixed in #469.

**Comment by cjblomqvist** [comment]
Excellent! Thanks for good work and sorry I've been totally jammed throughout the summer!
